modes/parkour/moderation.lua

```diff
@@ -0,0 +1,447 @@
+				return tfm.exec.chatMessage("<v>[#] <r>"..playerid.. " has permbanned.", player)
+			else
+				if not tonumber(minutes) then 
+					return tfm.exec.chatMessage(minutes .. " doesnt look like a number ", player)
+				end
+				minutes = os.time()+tonumber(minutes)*60*1000
+				schedule("ban_change", playerid, minutes)
+				tfm.exec.chatMessage("<v>[#] <r>" ..playerid.. " has banned for " ..(tonumber(args[2])).. " minutes", player)
+			end
+		end
+
+	elseif cmd == "addmap" or cmd == "removemap" then -- !addmap code [low/high] or !removemap code
+		if not ranks.admin[player] and not ranks.mapper[player] and not ranks.manager[player] then
+			return
+		end
+
+		if quantity < 1 then 
+			return translatedChatMessage("invalid_syntax", player)
+		end
+
+		showChatCommands(player, cmd, args)
+
+		local mapcode = args[1]
+
+		if not tonumber(mapcode) then
+			mapcode = mapcode:gsub("^@", "")
+		end
+
+		for i=1, #to_do do
+			if tonumber(to_do[i][2]) == tonumber(mapcode) then
+				return tfm.exec.chatMessage("<v>[#] <r>Please wait for a minute before taking any action with the same map.", player)
+			end
+		end
+
+		if cmd == "addmap" then
+			local rotation = args[2]
+
+			if in_table(tonumber(mapcode), maps.list_low) or in_table(tonumber(mapcode), maps.list_high) then
+				tfm.exec.chatMessage("<v>[#] <r>Map @"..mapcode.." is already in rotation.", player)
+				return
+			end
+
+			if quantity < 2 then 
+				return tfm.exec.chatMessage("<v>[#] <r>!addmap [code] [low/high]", player)
+			end
+
+			if rotation ~= "low" and rotation ~= "high" then 
+				return tfm.exec.chatMessage("<v>[#] <r>Select a priority option: low, high", player)
+			end
+
+			if not tonumber(mapcode) then
+				return tfm.exec.chatMessage("<v>[#] <r>"..mapcode.." doesn't look like a map code!?", player)  
+			end
+
+			schedule(rotation .. "_map_change", tonumber(mapcode), true)
+			tfm.exec.chatMessage("<v>[#] <r>Map @"..mapcode.." added to the "..rotation.." priority list.", player)
+		else
+			mapcode = tonumber(mapcode)
+			if quantity < 1 then 
+				return tfm.exec.chatMessage("<v>[#] <r>!removemap [code]", player)
+			end
+
+			if quantity > 2 then -- bulk remove
+				for i = 1, #args do
+					if not tonumber(args[i]) then
+						args[i] = args[i]:gsub("^@", "")
+					end
+				end
+
+				for y = 1, #args do 
+					if in_table(tonumber(args[y]), maps.list_low) then
+						schedule("low_map_change", tonumber(args[y]), false)
+						tfm.exec.chatMessage("<v>[#] <r>Map @"..args[y].." removed from the low priority list.", player)
+						
+					elseif in_table(tonumber(args[y]), maps.list_high) then
+						schedule("high_map_change", tonumber(args[y]), false)
+						tfm.exec.chatMessage("<v>[#] <r>Map @"..args[y].." removed from the high priority list.", player)
+
+					else
+						tfm.exec.chatMessage("<v>[#] <r>Map @"..args[y].." is not found in any map priority list.", player)
+					end
+				end
+			return end
+
+			if in_table(mapcode, maps.list_low) then
```

Comment:
Careful: The list might be outdated.
I'd recommend simply doing this check whenever the maps file loads.

Reply:
(and simply merge both low_map_change and high_map_change, the file will determine whether that happens on low or high priority)